mergify_engine/context.py

```diff
@@ -423,22 +416,66 @@ async def get_mergify_config_file(self) -> typing.Optional[MergifyConfigFile]:
         if mergify_config_file is not cache.Unset:
             return mergify_config_file
 
-        config_location_cache = self.get_config_location_cache_key(
-            self.installation.owner_login, self.repo["name"]
+        cached_config_file = await self.get_cached_config_file(
+            self.repo["id"],
         )
-        cached_filename = await self.installation.redis.get(config_location_cache)
+
+        if cached_config_file is not None:
+            self._caches.mergify_config_file.set(cached_config_file)
+            return cached_config_file
+
+        config_file_cache_key = self.get_config_file_cache_key(self.repo["id"])
+        config_location_cache_key = self.get_config_location_cache_key(self.repo["id"])
+        cached_filename = await self.installation.redis.get(config_location_cache_key)
+        pipeline = await self.installation.redis.pipeline()
+
         async for config_file in self.iter_mergify_config_files(
             preferred_filename=cached_filename
         ):
+
             if cached_filename != config_file["path"]:
-                await self.installation.redis.set(
-                    config_location_cache, config_file["path"], ex=60 * 60 * 24 * 31
+                await pipeline.set(
+                    config_location_cache_key, config_file["path"], ex=60 * 60 * 24 * 31
                 )
+
+            await pipeline.set(
+                config_file_cache_key,
+                {
+                    "type": config_file["type"],
+                    "content": config_file["content"],
+                    "path": config_file["path"],
+                    "sha": config_file["sha"],
+                },
+                ex=60 * 60 * 24 * 31,
+            )
             self._caches.mergify_config_file.set(config_file)
+            await pipeline.execute()
             return config_file
 
-        await self.installation.redis.delete(config_location_cache)
+        await pipeline.delete(config_file_cache_key)
         self._caches.mergify_config_file.set(None)
+        await pipeline.execute()
+        return None
+
+    async def get_cached_config_file(
+        self,
+        repo_id: github_types.GitHubRepositoryIdType,
+    ) -> typing.Optional[MergifyConfigFile]:
+        config_file = await self.installation.redis.get(
+            self.get_config_file_cache_key(repo_id),
+        )
+
+        if config_file is not None:
+            jsonified_config_file = json.loads(config_file.replace("'", '"'))
```

Comment:
what is this replace sorcery?

Reply:
Ahah ! This sorcery is because Redis, when storing a `dict` is replacing somehow all the `"` with `'`, when we `redis.get()` the result is an `str` of the `dict` without any `"`, since `json.loads()` can't work without a properly formatted string, I needed to replace the `'`. But I'm sure there is a better solution, maybe formatting better before using `redis.set()` ?

Reply:
Did you serialize it in redis?

Reply:
Nope, did not serialised the dict format before `redis.set()`

Reply:
You need to call `json.dumps`!